admin/listPurchases/listView.html

```diff
@@ -77,13 +88,142 @@
             tbody.innerHTML = "<tr><td colspan='1000'>Fejl ved load af ordrer. Prøver igen automatisk. Alternativt kan du forsøge at opdatere siden (tryk F5).</td></tr>";
         }
         else {
-            renderEntriesToList(entries.filter(option.filter));
+            currentPageIndex = 0;
+            currentOption = option;
+            updateContent();    
         }
-
         return false;
     });
 });
 
+const amountOfShownSubpages = 5;
+const entriesPerSubpage = 20;
+let amountOfSubpages;
+
+function generateSubPageNavigation(){
+    if(orderSearchBar.value){
+        amountOfSubpages = Math.ceil(entries.filter(currentOption.filter).filter(searchBarFilter).length / entriesPerSubpage);
+    }else{
+        amountOfSubpages = Math.ceil(entries.filter(currentOption.filter).length / entriesPerSubpage);
+    }
+
+    if(currentPageIndex >= amountOfSubpages){
+        currentPageIndex = amountOfSubpages - 1;
+    } 
+
+    let startIndexToRender = 0;
+    let subpageNavigation = [];
+
+    const extraNavNeeded = amountOfSubpages > amountOfShownSubpages;
+
+    if(extraNavNeeded){
+        subpageNavigation.push(`<a href="#" class="option subpage-selector subpage-selector-to-start">&lt&lt</a>`);
+        subpageNavigation.push(`<a href="#" class="option subpage-selector subpage-selector-backwards">&lt</a>`);
+
+        startIndexToRender = currentPageIndex;
+        if(startIndexToRender + Math.floor(amountOfShownSubpages/2) >= amountOfSubpages){
+            startIndexToRender = amountOfSubpages - amountOfShownSubpages;
+        }else if(startIndexToRender - Math.floor(amountOfShownSubpages/2) > 0){
+            startIndexToRender -= Math.floor(amountOfShownSubpages/2);
```

Comment:
You do the same math here (`Math.floor(...)`) several times. Can you guess what I'm going to suggest? :smile: